src/components/PopupWithForm.js

```diff
@@ -32,13 +37,11 @@ export default class PopupWithForm extends Popup {
   // event listener to the form and call the setEventListeners() method of the
   // parent class.
   setEventListeners() {
-    this._popupForm.addEventListener("submit", (evt) => {
-      evt.preventDefault();
+    super.setEventListeners();
+    this._popupForm.addEventListener("submit", () => {
       const formValues = this._getInputValues();
       this._handleFormSubmit(formValues);
       this._popupForm.reset();
     });
-    super.setEventListeners();
-    // this._popupForm.reset();
   }
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```

Reply:
Hey gennady-bars, quick question, don't we need to assign 'this._submitBtnText' to something passed though the constructor? 

With the method you've suggested I am not completely sure what we're setting 'this._submitBtnText' to.

Reply:
`this._submitBtnText = this._submitBtn.textContent`

Here we get the **initial button text** which is in `index.html`, so you don't need to know if it's `Yes`, `Save`, `Submit` or anything else